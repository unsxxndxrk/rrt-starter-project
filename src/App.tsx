import { MuiThemeProvider } from "@material-ui/core/styles";
import Routes from "Routes";
import { appTheme } from "theme";
import CssBaseline from "@material-ui/core/CssBaseline";
import { Router } from "react-router-dom";
import { appHistory } from "appHistory";
import { Provider } from "react-redux";
import store from "store";

const App: React.FC = () => {
  return (
    <Provider store={store}>
      <MuiThemeProvider theme={appTheme}>
        <CssBaseline />
        <Router history={appHistory}>
          <Routes />
        </Router>
      </MuiThemeProvider>
    </Provider>
  );
};

export default App;
