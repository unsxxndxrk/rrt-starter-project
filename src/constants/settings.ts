export const APP_DEFAULT_LANGUAGE: string = "en";
export const API_ENDPOINT: string =
  process.env.REACT_APP_API_ENDPOINT || window.location.origin;
