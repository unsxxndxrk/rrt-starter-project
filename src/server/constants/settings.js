const ORIGIN = "http://localhost";
const PORT = "8000";
const USERS_PATH = "./data/users.json";
const TODOS_PATH = "./data/todos.json";

exports.ORIGIN = ORIGIN;
exports.PORT = PORT;
exports.USERS_PATH = USERS_PATH;
exports.TODOS_PATH = TODOS_PATH;
