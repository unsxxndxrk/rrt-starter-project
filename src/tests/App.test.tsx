import App from "App";
import { render } from "tests/settings/render";

describe("<App />", () => {
  test("should render App component", () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });
});
