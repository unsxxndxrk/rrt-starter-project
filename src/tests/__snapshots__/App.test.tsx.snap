// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render App component 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg"
    >
      <div
        class="MuiBox-root MuiBox-root makeStyles-root"
      >
        <div
          class="MuiGrid-root makeStyles-nav MuiGrid-container MuiGrid-spacing-xs-5 MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
        >
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <a
              href="/"
            >
              Home
            </a>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <a
              href="/examples"
            >
              Examples
            </a>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <a
              href="/about"
            >
              About
            </a>
          </div>
        </div>
        <div
          class="makeStyles-root"
        >
          <div
            class="makeStyles-langWrap"
          >
            <div
              class="makeStyles-lang active"
            >
              en
            </div>
            |
          </div>
          <div
            class="makeStyles-langWrap"
          >
            <div
              class="makeStyles-lang"
            >
              ru
            </div>
          </div>
        </div>
      </div>
      <hr
        class="MuiDivider-root"
      />
      <main>
        <div
          class="makeStyles-content"
        >
          <div
            class="MuiBox-root MuiBox-root"
          >
            <div
              class="makeStyles-logoWrap"
            >
              <img
                alt="rrt"
                src="RRT_MAIN.svg"
              />
            </div>
            <h1
              class="MuiTypography-root MuiTypography-h1 MuiTypography-alignCenter"
            >
              ReactJS, Redux, TypeScript
            </h1>
            <div
              class="makeStyles-textWrap"
            >
              <p
                class="MuiTypography-root makeStyles-text MuiTypography-body1 MuiTypography-alignCenter"
              >
                Starter project for developing user interfaces
              </p>
            </div>
            <div
              class="makeStyles-buttonBox"
            >
              <a
                href="/about"
              >
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiButton-label"
                  >
                    About
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </a>
            </div>
          </div>
        </div>
      </main>
    </div>
  </div>
</DocumentFragment>
`;
